digitalocean/spaces/resource_spaces_bucket_cors_configuration.go

```diff
@@ -0,0 +1,259 @@
+	}
+
+	bucket := d.Get("bucket").(string)
+	rawCors := d.Get("cors_rule").([]interface{})
+
+	if len(rawCors) == 0 {
+		return diag.Errorf("Spaces bucket CORS must not be empty")
+	}
+
+	input := &s3.PutBucketCorsInput{
+		Bucket: aws.String(bucket),
+		CORSConfiguration: &s3.CORSConfiguration{
+			CORSRules: expandBucketCorsConfigurationCorsRules(rawCors, bucket),
+		},
+	}
+
+	log.Printf("[DEBUG] Trying to configure CORS for Spaces bucket: %s", bucket)
+	_, err = conn.PutBucketCorsWithContext(ctx, input)
+
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "NoSuchKey" {
+			return diag.Errorf("Unable to configure CORS for Spaces bucket because the bucket does not exist: '%s'", bucket)
+		}
+		return diag.Errorf("Error occurred while configuring CORS for Spaces bucket: %s", err)
+	}
+
+	d.SetId(bucket)
+	return resourceDigitalOceanBucketPolicyRead(ctx, d, meta)
+}
+
+func resourceDigitalOceanBucketCorsConfigurationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn, err := s3connFromSpacesBucketCorsResourceData(d, meta)
+	if err != nil {
+		return diag.Errorf("Error occurred while fetching Spaces bucket CORS configuration: %s", err)
+	}
+
+	log.Printf("[DEBUG] Trying to fetch Spaces bucket CORS configuration for bucket: %s", d.Id())
+	response, err := conn.GetBucketCorsWithContext(ctx, &s3.GetBucketCorsInput{
+		Bucket: aws.String(d.Id()),
+	})
+
+	if err != nil {
+		return diag.Errorf("Error occurred while fetching Spaces bucket CORS configuration: %s", err)
+	}
+
+	d.Set("bucket", d.Id())
+
+	if err := d.Set("cors_rule", flattenBucketCorsConfigurationCorsRules(response.CORSRules)); err != nil {
+		return diag.Errorf("setting cors_rule: %s", err)
+	}
+
+	return nil
+}
+
+func resourceDigitalOceanBucketCorsConfigurationUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return resourceDigitalOceanBucketCorsConfigurationCreate(ctx, d, meta)
+}
+
+func resourceBucketCorsConfigurationDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn, err := s3connFromSpacesBucketCorsResourceData(d, meta)
+	if err != nil {
+		return diag.Errorf("Error occurred while deleting Spaces bucket CORS configuration: %s", err)
+	}
+
+	bucket := d.Id()
+
+	log.Printf("[DEBUG] Trying to delete Spaces bucket CORS Configuration for bucket: %s", d.Id())
+	_, err = conn.DeleteBucketCorsWithContext(ctx, &s3.DeleteBucketCorsInput{
+		Bucket: aws.String(bucket),
+	})
+
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "BucketDeleted" {
+			return diag.Errorf("Unable to remove Spaces bucket CORS configuration because bucket '%s' is already deleted", bucket)
+		}
+		return diag.Errorf("Error occurred while deleting Spaces Bucket CORS configuration: %s", err)
+	}
+	return nil
+
+}
+
+func expandBucketCorsConfigurationCorsRules(rawCors []interface{}, bucketName string) []*s3.CORSRule {
```

Comment:
Good catch. I ended up using a different method: `expandBucketCorsConfigurationCorsRules` because the `cors_rule` in `digitalocean_spaces_bucket` is of a different type than the `cors_rule` in `digitalocean_spaces_bucket_cors_configuration`, similar to AWS [here](https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/s3/bucket_cors_configuration.go#L48)